src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -40,43 +50,48 @@ void register_smallAge_isNotOk() {
     }
 
     @Test
-    void register_existingLogin_isNotOk() {
-        user.setLogin(user.getLogin());
+    void register_nullLogin_isNotOk() {
+        user.setLogin(null);
         assertThrows(RuntimeException.class, () -> registrationServiceImpl.register(user));
     }
 
     @Test
-    void register_nullPassword_isNotOk() {
-        user.setPassword(null);
-        assertThrows(RuntimeException.class, () -> registrationServiceImpl.register(user));
+    void register_existingLogin_isNotOk() {
+        User registeredUser = storageDao.add(user);
+        Throwable thrown = assertThrows(RuntimeException.class, () -> {
+            registrationServiceImpl.register(user);
+        });
+        assertEquals("This login is already exist", thrown.getMessage());
     }
 
     @Test
-    void register_smallPassword_isNotOk() {
-        user.setPassword(INVALID_SMALL_PASSWORD);
-        assertThrows(RuntimeException.class, () -> registrationServiceImpl.register(user));
+    void register_nullPassword_isNotOk() {
+        user.setLogin(VALID_NEW_LOGIN);
+        user.setPassword(null);
+        Throwable thrown1 = assertThrows(RuntimeException.class, () -> {
+            registrationServiceImpl.register(user);
+        });
+        assertEquals("Password can't be null", thrown1.getMessage());
     }
 
     @Test
-    void register_notAllUserdataIsExcepted_isNotOk() {
-        user.setAge(VALID_AGE);
+    void register_smallPassword_isNotOk() {
+        user.setLogin(VALID_NEW_LOGIN);
         user.setPassword(INVALID_SMALL_PASSWORD);
-        assertThrows(RuntimeException.class, () -> registrationServiceImpl.register(user));
-    }
-
-    @Test
-    void register_nullLogin_isNotOk() {
-        user.setLogin(null);
-        assertThrows(RuntimeException.class, () -> registrationServiceImpl.register(user));
+        Throwable thrown = assertThrows(RuntimeException.class, () -> {
+            registrationServiceImpl.register(user);
+        });
+        assertEquals("Password length should be "
+                + MIN_PASSWORD_LENGTH + " or more symbols", thrown.getMessage());
     }
 
     @Test
-    void register_validUser_isOk() {
-        user.setPassword(VALID_PASSWORD);
-        user.setAge(VALID_AGE);
-        user.setLogin(VALID_NEW_LOGIN);
-        User registeredUser = storageDao.add(user);
-        assertEquals(user, registeredUser);
-        assertEquals(1, Storage.people.size());
+    void register_testUser_isOk() {
+        User testUser = new User();
+        testUser.setAge(VALID_AGE);
+        testUser.setPassword(VALID_PASSWORD);
+        testUser.setLogin(VALID_NEW_LOGIN);
+        User registeredUser = registrationServiceImpl.register(testUser);
+        assertEquals(2, Storage.people.size());
     }
```

Comment:
can we use `user` field here?
also it would be better to clear our `Storage.people` before each test
also we can use `storageDao.get` to check if we have user in our storage